src/browser-realm.ts

```diff
@@ -1,134 +1,8 @@
-    let record: CachedReferencesRecord | undefined = WeakMapGet(cachedGlobalMap, global);
-    if (!isUndefined(record)) {
-        return record;
-    }
-    record = ObjectCreate(null) as CachedReferencesRecord;
-    // caching references to object values that can't be replaced
-    // window -> Window -> WindowProperties -> EventTarget
-    record.window = global.window;
-    record.document = global.document;
-    record.WindowProto = ReflectGetPrototypeOf(record.window);
-    record.WindowPropertiesProto = ReflectGetPrototypeOf(record.WindowProto);
-    record.EventTargetProto = ReflectGetPrototypeOf(record.WindowPropertiesProto);
-    record.DocumentProto = ReflectGetPrototypeOf(record.document);
-
-    // caching descriptors
-    record.windowDescriptors = getOwnPropertyDescriptors(record.window);
-    record.WindowProtoDescriptors = getOwnPropertyDescriptors(record.WindowProto);
-    record.WindowPropertiesProtoDescriptors = getOwnPropertyDescriptors(record.WindowPropertiesProto);
-    record.EventTargetProtoDescriptors = getOwnPropertyDescriptors(record.EventTargetProto);
-
-    return record;
-}
-
-/**
- * Initialization operation to capture and cache all unforgeable references
- * and their respective descriptor maps before any other code runs, this
- * usually help because this library runs before anything else that can poison
- * the environment.
- */
-getCachedReferences(globalThis);
-
-/**
- * global descriptors are a combination of 3 set of descriptors:
- * - first, the key of the red descriptors define the descriptors
- *   provided by the browser when creating a brand new window.
- * - second, once we know the base keys, we get the actual descriptors
- *   from the blueDescriptors, since those are the one we want to provide
- *   access to via the membrane.
- * - third, the user of this library can provide endowments, which define
- *   global descriptors that should be installed into the sandbox on top
- *   of the base descriptors.
- *
- * Note: The main reason for using redDescriptors as the base keys instead
- * of blueDescriptor is because there is no guarantee that this library is
- * the first one to be evaluated in the host app, which means it has no ways
- * to determine what is a real DOM API vs app specific globals.
- *
- * Quirk: The only quirk here is for the case in which this library runs
- * after some other code that patches some of the DOM APIs. This means
- * the installed proxy in the sandbox will point to the patched global
- * API in the blue realm, rather than the original, because we don't have
- * a way to access the original anymore. This should not be a deal-breaker
- * if the patched API behaves according to the spec.
- *
- * The result of this method is a descriptor map that contains everything
- * that will be installed (via the membrane) as global descriptors in
- * the red realm.
- */
-function aggregateGlobalDescriptors(
```

Comment:
moved to window.ts and renamed to be aggregateWindowDescriptors. It adds more unforgeables, and prevents intrinsics override.